src/LanguageServer/Protocol/Features/Diagnostics/EngineV2/DiagnosticIncrementalAnalyzer.CompilationManager.cs

```diff
@@ -59,8 +61,8 @@ internal partial class DiagnosticAnalyzerService
         // </summary>
         async Task<CompilationWithAnalyzersPair?> CreateCompilationWithAnalyzersAsync()
         {
-            var projectAnalyzers = stateSets.SelectAsArray(s => !s.IsHostAnalyzer, s => s.Analyzer);
-            var hostAnalyzers = stateSets.SelectAsArray(s => s.IsHostAnalyzer, s => s.Analyzer);
+            var projectAnalyzers = analyzers.WhereAsArray(static (s, info) => !info.IsHostAnalyzer(s), hostAnalyzerInfo);
```

Comment:
>var projectAnalyzers = analyzers.WhereAsArray(static (s, info) => !info.IsHostAnalyzer(s), hostAnalyzerInfo); [](http://example.com/codeflow?start=12&length=109)

mentioned this after one of the earlier PRs had already been merged, but the projectAnalyzers/hostAnalyzers arrays appear to be intermediary arrays used only once, and the WhereAsArray conditions to create the filtered versions could be slightly modified to not need the intermediary arrays.

Reply:
yup.  can likely go through improve more here.